pandas/core/generic.py

```diff
@@ -2493,10 +2493,11 @@ def to_sql(self, name, con, schema=None, if_exists='fail', index=True,
         chunksize : int, optional
             Rows will be written in batches of this size at a time. By default,
             all rows will be written at once.
-        dtype : dict, optional
-            Specifying the datatype for columns. The keys should be the column
-            names and the values should be the SQLAlchemy types or strings for
-            the sqlite3 legacy mode.
+        dtype : dict or a SQLAchemy type, optional
```

Comment:
```suggestion
        dtype : dict or scalar, optional
```

Reply:
Thanks. I wasn't aware of this use of `scalar`. Just to double check, `pandas.api.types.is_scalar()` returns `False:

```
from pandas.api.types import is_scalar
from sqlalchemy.types import Integer
is_scalar(Integer) # False
```